pyanaconda/localization.py

```diff
@@ -148,20 +148,20 @@ def langcode_matches_locale(langcode, locale):
     :rtype: bool
 
     """
-
-    langcode_parts = parse_langcode(langcode)
-    locale_parts = parse_langcode(locale)
-
-    if not langcode_parts or not locale_parts:
+    if not is_valid_langcode(langcode) or not is_valid_langcode(locale):
         # to match, both need to be valid langcodes (need to have at least
         # language specified)
         return False
 
+    langcode_parsed = langtable.parse_locale(langcode)
+    locale_parsed = langtable.parse_locale(locale)
+
     # Check parts one after another. If some part appears in the langcode and
     # doesn't match the one from the locale (or is missing in the locale),
     # return False, otherwise they match
     for part in ("language", "territory", "script", "encoding"):
-        if langcode_parts[part] and langcode_parts[part] != locale_parts.get(part):
+        value = langcode_parsed.__getattribute__(part)
+        if value and value!= locale_parsed.__getattribute__(part):
```

Comment:
Please change the `__getattribute__` to [getattr()](https://docs.python.org/3/library/functions.html#getattr) calls.

Reply:
Will do!